src/components/protocol_handler/src/protocol_handler_impl.cc

```diff
@@ -255,6 +297,27 @@ void ProtocolHandlerImpl::SendStartSessionAck(
       strncpy(protocolVersionString, (*minVersion).to_string().c_str(), 255);
       bson_object_put_string(
           &payloadObj, "protocolVersion", protocolVersionString);
+    } else if (serviceTypeValue == kMobileNav && additional_params != NULL) {
```

Comment:
I think that this `additional_params` value could easily be used to house any service-specific params. Plus we may be able to just add the generic parameters to this object instead of creating a new object.